src/main/java/org/jetbrains/research/groups/ml_methods/refactoring/detection/RMiner.java

```diff
@@ -0,0 +1,99 @@
+
+        String tmpDir = System.getProperty("java.io.tmpdir");
+        Repository repository = gitService.cloneIfNotExists(
+                tmpDir + "/" + projectName,
+                repositoryUrl.toString());
+
+        List<MoveMethodRefactoringFromVCS> detectedRefactorings = new ArrayList<>();
+        miner.detectAll(repository, "master", new RefactoringHandler() {
+            @Override
+            public void handle(RevCommit commitData, List<Refactoring> refactorings) {
+                System.out.print("...");
+                LOGGER.info("Processing new commit. Commit hash: " + commitData.getId().getName());
+                for (Refactoring refactoring : refactorings) {
+                    LOGGER.info("Refactoring: " + refactoring.toString());
+                    MoveOperationRefactoring moveOperationRefactoring = (MoveOperationRefactoring) refactoring;
+                    String targetClassQualifiedName = moveOperationRefactoring.getMovedOperation().getClassName();
+                    String originalClassQualifiedName = moveOperationRefactoring.getOriginalOperation().getClassName();
+                    String movedMethodName = moveOperationRefactoring.getMovedOperation().getName();
+                    String originalMethodName = moveOperationRefactoring.getOriginalOperation().getName();
+                    List<String> movedParamsClassesQualifiedNames =
+                            moveOperationRefactoring.getMovedOperation().getParametersWithoutReturnType().stream()
+                                    .map(umlParameter -> umlParameter.getType().toString())
+                                    .collect(Collectors.toList());
+                    List<String> originalParamsClassesQualifiedNames =
+                            moveOperationRefactoring.getOriginalOperation().getParametersWithoutReturnType().stream()
+                                    .map(umlParameter -> umlParameter.getType().toString())
+                                    .collect(Collectors.toList());
+                    detectedRefactorings.add(
+                            new MoveMethodRefactoringFromVCS(
+                                    repositoryUrl,
+                                    commitData.getId().getName(),
+                                    targetClassQualifiedName,
+                                    originalClassQualifiedName,
+                                    originalMethodName,
+                                    movedMethodName,
+                                    originalParamsClassesQualifiedNames,
+                                    movedParamsClassesQualifiedNames
+                            )
+                    );
+                }
+            }
+        });
+        return new DetectedRefactoringsInRepository(repositoryUrl, detectedRefactorings);
+    }
+
+    @NotNull
+    @Override
+    public List<DetectedRefactoringsInRepository> detect(@NotNull List<URL> repositoryUrls) throws Exception {
```

Comment:
Looks like a method that should be implemented in abstract class, as it is independent of underlying algorithm.

Reply:
I would say it's even better to move this code to main class (or somewhere near it) and add errors handling.

Reply:
Done. Moved to abstract and added error handling.